pages/extensions/amp_example_preview/util/example_extractor.py

```diff
@@ -0,0 +1,82 @@
+import re
+
+EXAMPLE_PATTERN = re.compile(r'\[\s*example(\s[^\]]*)?\](.*?\n```html *\n(.*?)\n```.*?)\[\s*/\s*example\s*\]',
+                             re.IGNORECASE | re.MULTILINE | re.DOTALL)
+
+PREVIEW_ATTRIBUTE_PATTERN = re.compile(r'\spreview\s*=\s*"([^"]+)"')
+PLAYGROUND_ATTRIBUTE_PATTERN = re.compile(r'\splayground\s*=\s*"([^"]+)"')
+IMPORT_ATTRIBUTE_PATTERN = re.compile(r'\simport\s*=\s*"([^"]+)"')
+TEMPLATE_ATTRIBUTE_PATTERN = re.compile(r'\stemplate\s*=\s*"([^"]+)"')
```

Comment:
Parsing these via regex is overkill (and harder to maintain). Instead parse these into a settings map using string split.

Reply:
I create an attribute map now, but still use regex to find the attributes.